synfig-studio/src/gui/dialogs/canvasresize.cpp

```diff
@@ -0,0 +1,297 @@
+/* === S Y N F I G ========================================================= */
+/*!	\file dialogs/canvasresize.cpp
+**	\brief Template File
+**
+**	$Id$
+**
+**	\legal
+**	Copyright (c) 2002-2005 Robert B. Quattlebaum Jr., Adrian Bentley
+**
+**	This file is part of Synfig.
+**
+**	Synfig is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 2 of the License, or
+**	(at your option) any later version.
+**
+**	Synfig is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with Synfig.  If not, see <https://www.gnu.org/licenses/>.
+**	\endlegal
+*/
+/* ========================================================================= */
+
+#ifdef USING_PCH
+#	include "pch.h"
+#else
+#ifdef HAVE_CONFIG_H
+#	include <config.h>
+#endif
+
+#include "canvasresize.h"
+
+#include <gui/app.h>
+#include <gui/dialogs/canvasproperties.h>
+#include <gui/localization.h>
+#include <gui/resourcehelper.h>
+#include <gui/widgets/widget_link.h>
+
+#endif
+
+using namespace etl;
+using namespace Gtk;
+using namespace sigc;
+using namespace std;
+using namespace studio;
+using namespace synfig;
+using namespace synfigapp;
```

Comment:
Can we specify only the needed directive instead of pulling all of them at once?
Like this
```c++
using etl::handle;
using Gtk::Window;
using Gtk::Spinbutton;
...
```

Reply:
Yes. I don't like to expose the whole namespace for everything, because it can could lead to long building times for large projects and files.
Yeah yeah, I'm old lol